packages/graphql/src/classes/Executor.ts

```diff
@@ -189,13 +197,28 @@ export class Executor {
             },
         };
     }
+    
+    private throwIfVersionMismatch(serverInfo: ServerInfo) {
+        if (this.neo4jDatabaseInfo) {
+            const protocolVersion = serverInfo.protocolVersion;
+            if (protocolVersion) {
+                const versions = protocolVersion.toString().split(".").map(digit => parseInt(digit, 10));
+                const major = versions[0];
+                const minor = versions[1] ?? 0;
+                if (major !== this.neo4jDatabaseInfo.version.major || minor !== this.neo4jDatabaseInfo.version.minor) {
+                    throw new VersionMismatchError(major, minor);
+                }
+            }
+        }
+    }
 
     private async sessionRun(query: string, parameters, defaultAccessMode, session): Promise<QueryResult> {
         const transactionType = `${defaultAccessMode.toLowerCase()}Transaction`;
-        const result = await session[transactionType](
-            (transaction: Transaction) => this.transactionRun(query, parameters, transaction),
-            this.getTransactionConfig()
-        );
+        const result = await session[transactionType](async (transaction: Transaction) => {
+            const txResult = await this.transactionRun(query, parameters, transaction);
+            this.throwIfVersionMismatch(txResult.summary.server);
```

Comment:
I really think this is unnecessary - I don't think we should be worried if we get to this stage because the Cypher is likely to have executed without error.

Reply:
My idea was around the scenario when the version is automatically detected and later the database is changed or updated, in that case, we could rollback the transaction and catch the error at the level when the resolved is invoked, change the memorized version to the new one and re-generate the query using the newer version.
We could think of another mechanism to automatically update the version anyway. 
I think the rollback mechanism will become necessary if we're going to adopt a 'retry' handler, if not, we could remove both in my opinion!
Happy to remove both, we could instead just print a warning like "different Neo4j version detected".